analyzers/src/SonarAnalyzer.VisualBasic/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -53,5 +55,30 @@ public static bool IsPartOfBinaryNegationOrCondition(this SyntaxNode node)
 
         public static string FindStringConstant(this SyntaxNode node, SemanticModel semanticModel) =>
             FindConstantValue(node, semanticModel) as string;
+
+        public static ControlFlowGraph CreateCfg(this SyntaxNode block, SemanticModel semanticModel)
+        {
+            var operation = semanticModel.GetOperation(block);
+            var rootSyntax = operation.RootOperation().Syntax;
+            var cfg = ControlFlowGraph.Create(rootSyntax, semanticModel);
+            if (block is LambdaExpressionSyntax)
+            {
+                // We need to go up and track all possible enclosing lambdas and other FlowAnonymousFunctionOperations
+                var cfgFlowOperations = cfg.FlowAnonymousFunctionOperations();  // Avoid recomputing for ancestors that do not produce FlowAnonymousFunction
```

Comment:
In order to avoid recomputing we would need to transform the result into an array or a list. The current implementation does the evaluation multiple times.

Reply:
🤦‍♂️